datalad/support/repo.py

```diff
@@ -249,3 +249,41 @@ class RepoInterface(object):
 
     # Test!
     pass
+
+
+def path_based_str_repo(cls):
```

Comment:
There is still a repo->repr typo here.

Reply:
Pushed  a fixup.

<details>
<summary>range-diff</summary>

```
1:  1f84216366 ! 1:  0c61b58457 RF: @path_based_str_repo defining __str__ and __repr__
    @@ Metadata
     Author: Yaroslav Halchenko <debian@onerussian.com>
     
      ## Commit message ##
    -    RF: @path_based_str_repo defining __str__ and __repr__
    +    RF: @path_based_str_repr defining __str__ and __repr__
    +
    +    Modified-by: Kyle Meyer <kyle@kyleam.com>
    +      Fix repo->repr typo.
     
      ## datalad/distribution/dataset.py ##
     @@
      from datalad.support.exceptions import (
          NoDatasetFound,
      )
    -+from datalad.support.repo import path_based_str_repo
    ++from datalad.support.repo import path_based_str_repr
      from datalad.support.gitrepo import (
          GitRepo,
      )
    @@ datalad/distribution/dataset.py
      lgr.log(5, "Importing dataset")
      
      
    -+@path_based_str_repo
    ++@path_based_str_repr
      class Dataset(object, metaclass=PathBasedFlyweight):
          """Representation of a DataLad dataset/repository
      
    @@ datalad/support/gitrepo.py
          PathBasedFlyweight,
     -    RepoInterface
     +    RepoInterface,
    -+    path_based_str_repo,
    ++    path_based_str_repr,
      )
      
      # shortcuts
    @@ datalad/support/gitrepo.py: def _from_line(cls, line):
      Submodule = namedtuple("Submodule", ["name", "path", "url"])
      
      
    -+@path_based_str_repo
    ++@path_based_str_repr
      class GitRepo(RepoInterface, metaclass=PathBasedFlyweight):
          """Representation of a git repository
      
    @@ datalad/support/repo.py: class RepoInterface(object):
          pass
     +
     +
    -+def path_based_str_repo(cls):
    ++def path_based_str_repr(cls):
     +    """A helper decorator for a class to define str and repr based on its .path
     +
     +    For the rationale/discussion on why to bother distinguishing the two is
2:  60a5f94ff7 ! 2:  911da5c728 OPT: memoize __str__ and __repr__ of @path_based_str_repo"ed classes
    @@ Metadata
     Author: Yaroslav Halchenko <debian@onerussian.com>
     
      ## Commit message ##
    -    OPT: memoize __str__ and __repr__ of @path_based_str_repo"ed classes
    +    OPT: memoize __str__ and __repr__ of @path_based_str_repr"ed classes
     
         no check is done that .path was not changed -- so assumption is that it does
         not change.  Shaves off about 80% time on subsequent invocations
     
      ## datalad/support/repo.py ##
    -@@ datalad/support/repo.py: def path_based_str_repo(cls):
    +@@ datalad/support/repo.py: def path_based_str_repr(cls):
          # 62 ns ± 0.345 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
          # and similarly 58ns vs 97ns for %r vs !r
          def __str__(self):
```

</details>

Reply:
heh heh -- I just fixed in the docstring and missed that function itself had it. Thank you for fixing it!